es/baseapi.py

```diff
@@ -17,25 +21,78 @@ class Type(object):
 def check_closed(f):
     """Decorator that checks if connection/cursor is closed."""
 
-    def g(self, *args, **kwargs):
+    def wrap(self, *args, **kwargs):
         if self.closed:
             raise exceptions.Error(
                 "{klass} already closed".format(klass=self.__class__.__name__)
             )
         return f(self, *args, **kwargs)
 
-    return g
+    return wrap
 
 
 def check_result(f):
     """Decorator that checks if the cursor has results from `execute`."""
 
-    def g(self, *args, **kwargs):
+    def wrap(self, *args, **kwargs):
         if self._results is None:
             raise exceptions.Error("Called before `execute`")
         return f(self, *args, **kwargs)
 
-    return g
+    return wrap
+
+
+def get_type(data_type):
+    type_map = {
+        "text": Type.STRING,
+        "keyword": Type.STRING,
+        "integer": Type.NUMBER,
+        "half_float": Type.NUMBER,
+        "scaled_float": Type.NUMBER,
+        "geo_point": Type.STRING,
+        # TODO get a solution for nested type
+        "nested": Type.STRING,
+        "object": Type.STRING,
+        "date": Type.DATETIME,
+        "datetime": Type.DATETIME,
+        "short": Type.NUMBER,
+        "long": Type.NUMBER,
+        "float": Type.NUMBER,
+        "double": Type.NUMBER,
+        "bytes": Type.NUMBER,
+        "boolean": Type.BOOLEAN,
+        "ip": Type.STRING,
+        "interval_minute_to_second": Type.STRING,
+        "interval_hour_to_second": Type.STRING,
+        "interval_hour_to_minute": Type.STRING,
+        "interval_day_to_second": Type.STRING,
+        "interval_day_to_minute": Type.STRING,
+        "interval_day_to_hour": Type.STRING,
+        "interval_year_to_month": Type.STRING,
+        "interval_second": Type.STRING,
+        "interval_minute": Type.STRING,
+        "interval_day": Type.STRING,
+        "interval_month": Type.STRING,
+        "interval_year": Type.STRING,
+    }
+    return type_map[data_type.lower()]
```

Comment:
Should these be defined so that shared types are in `baseapi.py`, and ES and OD specific types are defined in `elastic/api.py` and `opendistro/api.py` respectively?

Reply:
yes specific and overrides, I'll probably assume the oficial elastic is the base.
These are important also: https://github.com/preset-io/elasticsearch-dbapi/blob/master/es/basesqlalchemy.py#L221

Reply:
I think I'll skip this for now, since having the same source of truth on type mapping is not causing any problems